### PR TITLE
Ensure that calendar stays on current view/date across page impressions

### PR DESCRIPTION
**Pull request by Guntrisoft**
- Storing a cookie of the current view and date of each calendar
- Config for fullcalendar on initialisation gets overriden with values from cookie
- Remembers current defaultView/defaultDate settings for a year (across browser sessions)

### CODE REVIEW

**Comment by Guntrisoft** [comment]
@benbarnett

File: app/assets/javascripts/calendar.es6 [hunk 1]
```diff
@@ -14,14 +17,44 @@ class Calendar {
       height: 'auto',
       allDaySlot: false,
       firstDay: 1,
-      defaultDate: moment(el.data('default-date'))
+      defaultDate: moment(el.data('default-date')),
+      viewRender: (view) => {
```
Comment (benbarnett) [inline app/assets/javascripts/calendar.es6]: The issue with having the full method in the default config is that this'll get blown away if a `viewRender` is passed in with the config.

Might be worth splitting this out, calling the 'default' one, and then running the passed in `config.viewRender` after the fact (if defined).

File: app/assets/javascripts/calendar.es6 [hunk 2]
```diff
@@ -1,9 +1,12 @@
-/* global moment */
+/* global moment, cookie */
 
 'use strict';
 
 class Calendar {
   constructor(el, config = {}) {
+    this.cookieName = this.constructor.name;
```
Comment (benbarnett) [inline app/assets/javascripts/calendar.es6]: Worth having this as part of the config? In the case of multiple calendars on a page etc.

**Comment by AndrewVos** [comment]
This lgtm, though I wonder if `defaultDate` should be stored. What if the user is a few months ahead? It might be weird coming back to the calendar and being in the wrong month?

**Comment by Guntrisoft** [comment]
@AndrewVos I can see some negatives and positives to storing it!

If they are editing an appointment on a particular date in the future and come back to it, then losing where they were date wise might be annoying.

Maybe see how they go with it, and whether TPAS/TP note any issues?

**Comment by Guntrisoft** [comment]
@benbarnett this has evolved a bit since the last time you reviewed it, based on the feedback you gave.

**Comment by benlovell** [comment]
@Guntrisoft @AndrewVos We definitely need some means of directing them back to the right day / view of the calendar after batch modifying a bunch of appointments. Whether it's via this approach or not I don't know...

**Comment by AndrewVos** [comment]
Agree with that @Guntrisoft.

LGTM

**Pull request merged**